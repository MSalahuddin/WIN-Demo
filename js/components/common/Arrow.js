import React from 'react';
import { TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import { color as StyleColor, hitSlop } from '../../styles';
import { scale } from '../../platformUtils';

export const ARROW_DIRECTION = {
  LEFT: 'LEFT',
  RIGHT: 'RIGHT',
  UP: 'UP',
  DOWN: 'DOWN'
};

// The following unnecessary destruction is to address a eslint issue
// Expected newline after ":" with a    declaration-colon-newline-after multi-line declaration
const { transparent } = StyleColor;
const StyledArrow = styled.View`
  width: 0;
  height: 0;
  border-width: ${({ size }) => scale(size)};
  border-style: solid;
  border-radius: ${scale(2)};
  border-top-color: ${({ color, arrowDirection }) => (arrowDirection === ARROW_DIRECTION.DOWN ? color : transparent)};
  border-bottom-color: ${({ color, arrowDirection }) => (arrowDirection === ARROW_DIRECTION.UP ? color : transparent)};
  border-left-color: ${({ color, arrowDirection }) => (arrowDirection === ARROW_DIRECTION.RIGHT ? color : transparent)};
  border-right-color: ${({ color, arrowDirection }) => (arrowDirection === ARROW_DIRECTION.LEFT ? color : transparent)};
`;

const Arrow = ({ arrowDirection, color, size, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} hitSlop={hitSlop(20, 20)} testID="arrow">
      <StyledArrow arrowDirection={arrowDirection} color={color} size={size} />
    </TouchableOpacity>
  );
};

Arrow.defaultProps = {
  color: StyleColor.black,
  arrowDirection: ARROW_DIRECTION.DOWN,
  size: 8,
  onPress: () => {}
};

Arrow.propTypes = {
  arrowDirection: PropTypes.oneOf([
    ARROW_DIRECTION.UP,
    ARROW_DIRECTION.DOWN,
    ARROW_DIRECTION.LEFT,
    ARROW_DIRECTION.RIGHT
  ]),
  color: PropTypes.string,
  size: PropTypes.number,
  onPress: PropTypes.func
};

export default Arrow;
