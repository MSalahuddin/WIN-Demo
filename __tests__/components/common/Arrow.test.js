import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import Arrow from '../../../js/components/common/Arrow';

describe('Component Arrow', () => {
  test('renders correctly', () => {
    const output = render(<Arrow />);
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when arrowDirection="UP"', () => {
    const output = render(<Arrow arrowDirection="UP" />);
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when arrowDirection="LEFT"', () => {
    const output = render(<Arrow arrowDirection="LEFT" />);
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when arrowDirection="RIGHT"', () => {
    const output = render(<Arrow arrowDirection="RIGHT" />);
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when Arrow is pressed', () => {
    const onPress = jest.fn();
    const { getByTestId } = render(<Arrow onPress={onPress} />);
    fireEvent(getByTestId('arrow'), 'press');
    expect(onPress).toHaveBeenCalled();
  });
});
