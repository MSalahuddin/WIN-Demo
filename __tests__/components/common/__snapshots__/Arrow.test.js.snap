// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Arrow renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="arrow"
>
  <View
    arrowDirection="DOWN"
    color="#000000"
    size={8}
    style={
      Array [
        Object {
          "borderBottomColor": "transparent",
          "borderLeftColor": "transparent",
          "borderRadius": 3,
          "borderRightColor": "transparent",
          "borderStyle": "solid",
          "borderTopColor": "#000000",
          "borderWidth": 12,
          "height": 0,
          "width": 0,
        },
      ]
    }
  />
</View>
`;

exports[`Component Arrow renders correctly when arrowDirection="LEFT" 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="arrow"
>
  <View
    arrowDirection="LEFT"
    color="#000000"
    size={8}
    style={
      Array [
        Object {
          "borderBottomColor": "transparent",
          "borderLeftColor": "transparent",
          "borderRadius": 3,
          "borderRightColor": "#000000",
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderWidth": 12,
          "height": 0,
          "width": 0,
        },
      ]
    }
  />
</View>
`;

exports[`Component Arrow renders correctly when arrowDirection="RIGHT" 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="arrow"
>
  <View
    arrowDirection="RIGHT"
    color="#000000"
    size={8}
    style={
      Array [
        Object {
          "borderBottomColor": "transparent",
          "borderLeftColor": "#000000",
          "borderRadius": 3,
          "borderRightColor": "transparent",
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderWidth": 12,
          "height": 0,
          "width": 0,
        },
      ]
    }
  />
</View>
`;

exports[`Component Arrow renders correctly when arrowDirection="UP" 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="arrow"
>
  <View
    arrowDirection="UP"
    color="#000000"
    size={8}
    style={
      Array [
        Object {
          "borderBottomColor": "#000000",
          "borderLeftColor": "transparent",
          "borderRadius": 3,
          "borderRightColor": "transparent",
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderWidth": 12,
          "height": 0,
          "width": 0,
        },
      ]
    }
  />
</View>
`;
